tidy3d/components/simulation.py

```diff
@@ -2494,15 +2494,24 @@ def _projection_monitors_homogeneous(cls, val, values):
         structures = values.get("structures") or []
         total_structures = [structure_bg] + list(structures)
 
-        for monitor in val:
-            if isinstance(monitor, (AbstractFieldProjectionMonitor, DiffractionMonitor)):
-                mediums = Scene.intersecting_media(monitor, total_structures)
-                # make sure there is no more than one medium in the returned list
-                if len(mediums) > 1:
-                    raise SetupError(
-                        f"{len(mediums)} different mediums detected on plane "
-                        f"intersecting a {monitor.type}. Plane must be homogeneous."
-                    )
+        with log as consolidated_logger:
+            for monitor_ind, monitor in enumerate(val):
+                if isinstance(monitor, (AbstractFieldProjectionMonitor, DiffractionMonitor)):
+                    mediums = Scene.intersecting_media(monitor, total_structures)
+                    # make sure there is no more than one medium in the returned list
+                    if len(mediums) > 1:
+                        raise SetupError(
+                            f"{len(mediums)} different mediums detected on plane "
+                            f"intersecting a {monitor.type}. Plane must be homogeneous."
+                        )
+                    # if <=1 medium, check if the medium is spatially uniform
```

Comment:
Is it possible to have 0 media? That kinda feels like something is wrong too, unless we're not taking the simulation background into account, but I think we are?

In other words not sure if we can remove the `for mat in mediums` and / or raise a different error if `mediums` is empty.

Reply:
It shouldn't be possible. An error has been raised now if 0 media.